mobile/library/common/extensions/filters/http/network_configuration/filter.cc

```diff
@@ -110,18 +110,10 @@ NetworkConfigurationFilter::resolveProxy(Http::RequestHeaderMap& request_headers
         // so that, if the stream is canceled and the filter chain is torn down in the meantime,
         // we will fail to aquire the weak_ptr lock and won't execute any callbacks on the resolved
         // proxies.
-        if (auto caller_ptr = weak_self.lock()) {
-          Network::ProxySettingsConstSharedPtr proxy_settings =
-              Network::ProxySettings::create(proxies);
-          // We are currently in the main thread, so post the proxy resolution callback to the
-          // worker thread's (i.e. the Engine thread) dispatcher.
-          caller_ptr->decoder_callbacks_->dispatcher().post([&weak_self, proxy_settings]() {
-            // Again, the stream may have been canceled in the interim, so try aquiring the
-            // filter through its weak_ptr before proceding with the proxy resolution callback.
-            if (auto filter_ptr = weak_self.lock()) {
-              filter_ptr->onProxyResolutionComplete(proxy_settings);
-            }
-          });
+        if (auto filter_ptr = weak_self.lock()) {
+          // This call does not need to be posted on the dispatcher, because
+          // onProxyResolutionComplete posts continuing with decoding to the dispatcher already.
```

Comment:
Hm. I'm having a hard time understand quite what's going on here. The old comment says that this method (`resolveProxy`) is running on the main thread, and we need to run code on the Engine thread. This happens by posting a task to the callback's dispatcher.

Is that old comment wrong, and actually this method (`resolveProxy`) is already running on the engine thread? If so, I think perhaps the comment here should say something like. "This method is currently running on the Engine thread because ..."

But I'm quite likely just a bit confused this morning.

Reply:
It is definitely confusing :)

You're right, the old comment was wrong.  There's actually only one thread in Envoy Mobile.  So `resolveProxy` is run on the engine thread (the one EM thread) and onProxyResolutionComplete, when it's ready to continue with decoding, posts on the same thread's dispatcher, so main point here was that the extra dispatcher post was unnecessary.

I updated the comment to try to make it more clear.